### ISSUE TITLE
Redis connection option

### ISSUE THREAD

**Issue by victorhahn** [beekeeper-studio/beekeeper-studio#80]
Would love to see a Redis connection option similar to Table Plus. Great work, will be watching this project for updates!

**Comment by rathboma** [comment]
How do you query redis data in table-plus? Not with SQL?

**Comment by victorhahn** [comment]
My understanding is that it would use redis commands to query data, not SQL.

https://tableplus.com/blog/2018/06/best-redis-gui-client-tableplus.html

This feature along with JSONB support (currently outputs [Object Object]) would make our team extremely interested in using beekeeper for our projects!

**Comment by AqilCont** [comment]
I agree with the above comment. Redis is extremely useful to me, while being extremely popular on top of it and would be a game-changer for this app's usability.

**Comment by rathboma** [comment]
@victorhahn the [object object] output is fixed.

I'll need to think on the redis support. Hadn't initially thought of doing that yet, but I can see the value in supporting all the different datastores for sure

**Comment by yurikilian** [comment]
@rathboma can we contribute on this? Submiting a pr?

**Comment by rathboma** [comment]
Sure would love a PR on this as I won't have time to work on it for a while.

**Comment by DominusKelvin** [comment]
I was glad to see this discussion. I really think a Redis connection option would be useful. I love Bekeeper and I really want it to be the only GUI I use on my machine for databases and datastores. Great work team

**Comment by Zerotask** [comment]
Redis support would be great. I use often a combination of relational / NoSQL (mongodb) / key-value storage (redis). At the moment I use for MongoDB and Redis own docker container to access it locally. With this feature (and the already accepted mongodb support) I could actually drop that and simplify it by using just Beekeeper Studio. :)

**Comment by productdevbook** [comment]
I hope this support comes

**Comment by A-childs-encyclopedia** [comment]
Any updates?

**Comment by korablin** [comment]
bump

**Comment by Elysian10** [comment]
I wanted to suggest this feature as well. Is it still coming?

**Comment by AlexBa** [comment]
Need it too. TablePlus has it already.

**Comment by PaVeilleux** [comment]
I wish we had an update on this

**Comment by LeoWie93** [comment]
Redis support is already implemented for the Ultimate edition: https://github.com/beekeeper-studio/beekeeper-studio?tab=readme-ov-file#supported-databases 

This issue can be closed? @rathboma

**Comment by mrsimb** [comment]
@rathboma Im experimenting with adding Redis support right now. Its a bit rough, and Im still learning the database, so I will probably need help with handling of some UI features in the future. Right now im focused on getting the basic stuff working and cleaning up the code.
Regarding translating the key-value storage model to the SQL DB client UI, I think i'll modify JsonViewer (or add a separate component) to be able to show/edit not only json, but other Redis data types as well.

![Image](https://github.com/user-attachments/assets/ef0d738a-85ba-463b-b97c-85259003d817)

**Comment by rathboma** [comment]
Damn! That's great progress.

Honestly one of the reasons I've put it off so long is that Redis is so very different from SQL databases, or even something like Dynamo or MongoDB, so I've been a little scared of how many UI changes we'd need.

I think your plan to use the JSON viewer and possibly some sort of table-like structure probably makes the most sense. I think Tabulator might work better than expected.

**Comment by mrsimb** [comment]
![Image](https://github.com/user-attachments/assets/6f88d7eb-a50d-45b2-9b8b-c59ac60312fa)

Added query support that mimics redis-cli.
It's nice that i can return result of executing each line as an array of results.
Not pushing yet, there's much to cleanup and review, especially vibe-coded parts.
